lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -26,6 +26,168 @@
 }
 }
 
+// https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers
+// https://tc39.es/ecma402/#sec-canonicalizeunicodelocaleid
+vm::CallResult<std::u16string> canonicalizeLocaleId(const std::u16string &localeId, vm::Runtime *runtime) {
+  llvh::Optional<bcp47_parser::ParsedLocaleIdentifier> parserOpt = bcp47_parser::parseLocaleId(localeId);
```

Comment:
Long type names like this are a good candidate for `auto`, especially when the type is clear based on the right hand side.

```suggestion
  auto parserOpt = bcp47_parser::parseLocaleId(localeId);
```